library/CHART-TEMPLATE/templates/deployment.yaml

```diff
@@ -39,6 +39,12 @@ spec:
             - name: http
               containerPort: 80
               protocol: TCP
+          env:
+            - name: NRIA_LICENSE_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "common.licenseKey.secretName" . }}
```

Comment:
I like the fact that this is custom secret or not it is totally transparent for the user